cpp/src/strings/char_types/char_cases.cu

```diff
@@ -0,0 +1,268 @@
+/*
+ * Copyright (c) 2020, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <vector>
+#include <map>
+#include <algorithm>
+
+#include <cudf/cudf.h>
+#include <cudf/utilities/error.hpp>
+
+//
+namespace cudf
+{
+namespace strings
+{
+namespace detail
+{
+
+namespace {
+
+  struct special_case_mapping_in {
+    uint16_t num_upper_chars;
+    uint16_t upper[3];
+    uint16_t num_lower_chars;
+    uint16_t lower[3];
+  };
+  const std::vector<special_case_mapping_in> codepoint_mapping_in {
```

Comment:
```suggestion
  constexpr std::vector<special_case_mapping_in> codepoint_mapping_in {
```

Reply:
I don't think you can do this.  Constructor for vector isn't constexpr.

Reply:
So it isn't. Any reason for using a vector instead of a static array? Better yet, a `std::array`?